src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperOutput.cs

```diff
@@ -15,16 +15,14 @@ namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
     /// </summary>
     public class TagHelperOutput
```

Comment:
for future consideration (not this PR) but related to the `TagHelperContent` / `DefaultTagHelperContent` split introduced here: suggest we split `TagHelperOutput` along similar lines and get the `Generate*()` methods and perhaps `ContentSet` out of tag helper authors' faces.

this split could go further and move the `Generate*()` methods completely out of this class (or class hierarchy).  if `_isTagNameNullOrWhitespace` were exposed as a property (like `ContentSet`), these methods would not require any `private` information.  could move generation many places but I suggest the `RazorPage.WriteLiteral[To]([TextWriter,] TagHelperExecutionContext)` methods I proposed elsewhere.  in any case, this would pull apart the mix of responsibilities `TagHelperOutput` has today.